security-cs/My_API/My_API/Controllers/ValuesController.cs.newCode

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using System.Data;
+using System.Data.SqlClient;
+
+
+namespace My_API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ValuesController : ControllerBase
+    {
+        // GET api/values
+        [HttpGet]
+        public ActionResult<IEnumerable<string>> Get()
+        {
+            return new string[] { "value1", "value2" };
+        }
+
+        // GET api/values/5
+        [HttpGet("{id}")]
+        public ActionResult<string> Get(string id)
+        {
+            var connection = new SqlConnection();
+            try
+            {
+                connection.ConnectionString = "db info";
+                connection.Open();
+                var selectSql = string.Format("select from MyStuff where id='{0}';", id);

```

Comment:
It would be nice if this issue (or an additional issue) spanned multiple source files to show how you can trace through.

Reply:
I did as you suggested and introduced a new class that the controller uses to load the data. Now SQ shows the taint value being propagated across 2 files. Have a fresh look.